### DISCUSSION TITLE
create3 topics not showing up on ros2 topic list

### DISCUSSION THREAD

**Discussion by Illyvia** [iRobotEducation/create3_docs#575] [Questions] [answered]
### How are you connecting to your Create 3?

Wi-Fi and USB Ethernet (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Raspberry Pi 4/ ubuntu 22.04/humble/turtlebot 4 - VirtualBox ubuntu 22.04 VM/humble 

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.2.6

### Which RMW is your robot running?

FastRTPS

### Is the robot connected to a network? If so, what is the network type?

home network

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

No

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

The Create 3 is connected to the pi through wifi and through a usb c to usb c cable. I have attempted to do `ros2 topic list` but it's not working. I have tried it on a VM, normal ubuntu on a pi, nothing is working. The pi is running turtlebot4 os at the moment (I don't own a turtlebot4, I'm only using the os).

I have set the RMW and sourced it every time. I really don't know what I'm doing wrong.

**Comment by alsora** [comment]
Hi @TheInventor2023,
I see that you closed this issue.
Were you able to resolve the problem? If yes, how did you do it? I'm sure that your experience would be useful for other people.

(If you couldn't solve the problem, please let me know and I'll try to help)

**Reply by Illyvia** [reply, marked as answer]
Yes, I was able to solve the problem. I had forgotten about the other issue I had opened; I tried some things from that, but what helped was creating a unicast XML file, and setting the FastRTPS default profile to it.
